nomad/structs/config/reporting.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package config
+
+import "github.com/hashicorp/nomad/helper/pointer"
+
+func DefaultReporting() *Reporting {
+	return &Reporting{
+		&LicenseConfig{},
+	}
+}
+
+type LicenseConfig struct {
+	Enabled *bool `hcl:"enabled"`
+}
+
+func (lc *LicenseConfig) Copy() *LicenseConfig {
+	if lc == nil {
+		return nil
+	}
+
+	nlc := *lc
+	nlc.Enabled = pointer.Copy(lc.Enabled)
+
+	return &nlc
+}
+
+func (lc *LicenseConfig) Merge(b *LicenseConfig) *LicenseConfig {
+	if lc == nil {
+		return b
+	}
+
+	result := *lc
+
+	if b == nil {
+		return &result
+	}
+
+	if b.Enabled != nil {
+		result.Enabled = b.Enabled
+	}
+
+	return &result
+}
+
+type Reporting struct {
```

Comment:
```suggestion
type ReportingConfig struct {
```

It stutters a bit, but I think all other structs in this package end in `Config`.